eodhistoricaldata/src/adapter.ts

```diff
@@ -0,0 +1,43 @@
+import { Requester, Validator, AdapterError } from '@chainlink/external-adapter'
+import { ExecuteWithConfig, ExecuteFactory } from '@chainlink/types'
+import { makeConfig, DEFAULT_ENDPOINT } from './config'
+import { price } from './endpoint'
+
+const inputParams = {
+  endpoint: false,
+}
+
+export const execute: ExecuteWithConfig = async (request, config) => {
+  const validator = new Validator(request, inputParams)
+  if (validator.error) throw validator.error
+
+  Requester.logConfig(config)
+
+  const jobRunID = validator.validated.id
+  const endpoint = validator.validated.data.endpoint || DEFAULT_ENDPOINT
+
+  let result
+  switch (endpoint) {
+    case price.Name: {
+      result = await price.execute(config, request)
```

Comment:
```suggestion
      result = await price.execute(request, config)
```
A recent PR made ExecuteWithConfig a generic. (This applies across all EAs in this PR)